ctc_lib/container_tool_list_command.go

```diff
@@ -19,19 +19,52 @@ package ctc_lib
 import (
 	"errors"
 
+	"github.com/GoogleCloudPlatform/runtimes-common/ctc_lib/constants"
+
 	"github.com/GoogleCloudPlatform/runtimes-common/ctc_lib/util"
 	"github.com/spf13/cobra"
 )
 
 type ContainerToolListCommand struct {
 	*ContainerToolCommandBase
-	OutputList []interface{}
+	OutputList      []interface{}
 	// RunO Executes cobra.Command.Run and returns an List[Output]
 	RunO func(command *cobra.Command, args []string) ([]interface{}, error)
+	// When defined, StreamO Executes cobra.Command.Run and streams each item in the List as its added.
+	// This will ignore the RunO function.
+	StreamO func(command *cobra.Command, args []string)
```

Comment:
StreamO is used to stream objects on to StdOut as they get added for CommandList.
CommandList runs a command which returns an Object List.
Traditionally, the CommandList, would display output after the entire List is available. 
This would not be ideal Container-structure-Tests since as we would want to  stream results as they are availble.

Your follow up questions:
1. Should there be a std err stream or std out stream?
I dont think so. The reason being, ctc_lib.Log object writes the messages to StdOut and StdErr with Log.Info or Log.Error methods etc.
They appear as they are written. 
They would mingle with the Object stream, but it makes sense to see the Log.Error messages on the screen as you get in to errors.

2. Verbosity: The channel only applies to Objects. 
The ContainerToolListCommand.DefaultTemplate contains the template which gets applied to the Object.
Library Users can always define a TemplateFunc which will can use the ctc_lib.Verbosity flag value (which we already define in ctc_lib) and decide what fields you want to filter if Verbosity is set to Info VS Debug.
The Users, can also define their own flag like --quiet and create template Function which will not display nothing if the flag value is true.
And to answer your third question, yes it up to the person using the lib to define a template and flags which can display shorted values. We do have this use case container_structure_tests and it will serve as a good example for this.
I will submit a PR for it tomorrow morning and you can get an idea.

Reply:
Hit enter soon.. 
Let me know if the above makes sense. Open to feedback.